### DISCUSSION TITLE
Mult-tenant

### DISCUSSION THREAD

**Discussion by FELS-Zak** [pocketbase/pocketbase#97] [Q&A]
Does this support tenant data? How would you go about doing that?

**Comment by ganigeorgiev** [comment]
This should be already supported - you can have tenancy at users (see #94) and/or on collection level (aka. via user/rel fields).

Unlike Firebase, PocketBase Collection API rules are both access and query filters.

For example, let's say we want each user to have access only to its own `projects` records.
To do that, we can create a new collection `projects` with a `user` field named `owner`.
And then for the API rules we can define the following simple expression:
```
owner = @request.user.id
```
The above literally, translates to - return/match all records with `owner` equal to the logged in user.
The user field also supports linking more than 1 user (aka. multiple owners).

---

Furthermore, you can even query nested relation fields (aka. over multiple tables). Let's say we have 2 collections with the following fields:

| projects             |
| -------------------- |
| title    (text field) |
| owner (user field) |

| sub_projects           |
| ---------------------- |
| title     (text field) |
| project (rel field)  |

To allow access to only to `sub_projects` that are linked to `projects` owned by the current logged in user, we can use the following API rule in the `sub_projects` collection:

```
project.owner = @request.user.id
```

---

You also have an option to query fields of non directly related collections (aka. joins) via the `@collection.NAME.FIELD.[SUBFIELD.*]`

More info you could find at https://pocketbase.io/docs/manage-collections/#rules-filters-syntax and https://pocketbase.io/docs/manage-users/

**Reply by jimafisk** [reply]
What about a scenario where someone wants to provide PB as a Service? The above example seems to be multi-tenancy for your custom application (which is awesome), but what if you want to provide multiple PB admin dashboards running off the same instance?

**Reply by ganigeorgiev** [reply]
I don't think this is supported at the moment. All admin accounts acts as "super admin" for the application (aka. each admin can access and modify everything). Access controls and permissions apply only for the app users.

If you want to have multiple different PB admin dashboards (each with its own database), you could:
- start the same binary multiple times with different ports and data directory
- OR create your own Go application and start multiple PocketBase services each in its own go routine (the only problem to this approach is that the `serve` command arguments currently can be passed only via cli, aka. you'll have to modidy `os.Args[]`; I may add in the future an option to set default values with some method, eg `pb.DefaultHttp(...)`, similar to the current `pb.DefaultDataDir()`).

**Reply by FELS-Zak** [reply]
Yep - this is also what I'm after .

**Reply by MedLabs** [reply]
for my side project I'm running ./pocketbase multi times with different --dir and different --http so every user uses its independant instance.
I don't know if this could affect server performance or may get harder to manage when there are hundreds of independent instances...
